packages/dev/core/src/Gizmos/scaleGizmo.ts

```diff
@@ -339,26 +364,32 @@ export class ScaleGizmo extends Gizmo implements IScaleGizmo {
      * Disposes of the gizmo
      */
     public dispose() {
-        [this.xGizmo, this.yGizmo, this.zGizmo, this.uniformScaleGizmo].forEach((gizmo) => {
+        const allGizmos = this.allGizmos;
+        for (let i = 0; i < 4; i++) {
+            const gizmo = allGizmos[i];
             if (gizmo) {
                 gizmo.dispose();
             }
-        });
-        this._observables.forEach((obs) => {
-            this.gizmoLayer.utilityLayerScene.onPointerObservable.remove(obs);
-        });
+        }
+        const { _observables, _uniformScalingMesh, _octahedron } = this;
+        const _obserLength = _observables.length;
+        const onPointerObservable = this.gizmoLayer.utilityLayerScene.onPointerObservable;
+        for (let i = 0; i < _obserLength; i++) {
+            const obs = _observables[i];
+            onPointerObservable.remove(obs);
+        }
         this.onDragStartObservable.clear();
         this.onDragObservable.clear();
         this.onDragEndObservable.clear();
-        [this._uniformScalingMesh, this._octahedron].forEach((msh) => {
-            if (msh) {
-                msh.dispose();
-            }
-        });
-        [this._coloredMaterial, this._hoverMaterial, this._disableMaterial].forEach((matl) => {
+        _uniformScalingMesh && _uniformScalingMesh.dispose();
+        _octahedron && _octahedron.dispose();
+
+        const materArr = [this._coloredMaterial, this._hoverMaterial, this._disableMaterial];
+        for (let i = 0; i < 3; i++) {
+            const matl = materArr[i];

```

Comment:
IMO, all the code changes in this method are not clearer and don't need optimization. We are in a dispose, after all, which is not called thousand of times a frame.
What may be considered clearer is:
```typescript
this._uniformScalingMesh?.dispose();
this._octahedron?.dispose();
```
instead of:
```typescript
[this._uniformScalingMesh, this._octahedron].forEach((msh) => {
    if (msh) {
        msh.dispose();
    }
});
```
(same thing for `_coloredMaterial`, `_hoverMaterial` and `_disableMaterial`)